src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -2561,7 +2561,19 @@ module Queries = struct
           if not with_seq_no then Error "Missing sequence information"
           else Ok (Itn_logger.get_logs start_log_id) )
 
-    let commands = [ auth; slots_won; internal_logs ]
+    let zkapp_cmd_limit =
+      field "zkAppCommandLimit"
+        ~args:
+          Arg.[ arg "limit" ~doc:"ZkApp commands per block limit." ~typ:int ]
+        ~typ:int
+        ~doc:"Set zkApp commands per block limit for the block producer."
+        ~resolve:(fun { ctx = _with_seq_no, mina; _ } () limit ->
+          let open Mina_lib.Config in
+          let config = Mina_lib.config mina in
+          config.zkapp_cmd_limit <- limit ;
```

Comment:
Mutating the value in the configuration won't update the value that you passed into the block producer when it was started. This is why I had suggested that you need a reference in my last review.

Reply:
Thanks, @nholland94. Do you have a suggestion as to where the ref should go?

Reply:
I would suggest it lives in a layer deeper than `Mina_lib`, perhaps directly in the block producer module, where it is passed into the staged ledger from. If we do wish to keep it in `Mina_lib`, we could pass the reference into the block producer from there. It just can't be a mutable record field since those are only mutable on the record itself, and are copied from the record when accessed as a value.

Reply:
Also, if we move the key to the block_producer module does the mina_graphql have access to this module to update the variable? Thanks.

Reply:
You can still configure it for different build profiles if it's in the block_producer module. Just initialize it like `let zkapps_limit = ref Mina_compile_config.zkapps_limit`.

Reply:
You can expose the ref from the Block_producer module and set it from graphql. If you want to hide the ref, you can expose a function `val set_zkapps_limit : int -> unit` instead.